docker/models/containers.py

```diff
@@ -0,0 +1,144 @@
+        )
+
+    def export(self, *args, **kwargs):
+        return self.client.api.export(self.id, *args, **kwargs)
+
+    def get_archive(self, *args, **kwargs):
+        return self.client.api.get_archive(self.id, *args, **kwargs)
+
+    def kill(self, *args, **kwargs):
+        return self.client.api.kill(self.id, *args, **kwargs)
+
+    def logs(self, *args, **kwargs):
+        return self.client.api.logs(self.id, *args, **kwargs)
+
+    def pause(self, *args, **kwargs):
+        return self.client.api.pause(self.id, *args, **kwargs)
+
+    def put_archive(self, *args, **kwargs):
+        return self.client.api.put_archive(self.id, *args, **kwargs)
+
+    def remove(self, *args, **kwargs):
+        return self.client.api.remove_container(self.id, *args, **kwargs)
+
+    def rename(self, *args, **kwargs):
+        return self.client.api.rename(self.id, *args, **kwargs)
+
+    def resize(self, *args, **kwargs):
+        return self.client.api.resize(self.id, *args, **kwargs)
+
+    def restart(self, *args, **kwargs):
+        return self.client.api.restart(self.id, *args, **kwargs)
+
+    def start(self, *args, **kwargs):
+        return self.client.api.start(self.id, *args, **kwargs)
+
+    def stats(self, *args, **kwargs):
+        return self.client.api.stats(self.id, *args, **kwargs)
+
+    def stop(self, *args, **kwargs):
+        return self.client.api.stop(self.id, *args, **kwargs)
+
+    def top(self, *args, **kwargs):
+        return self.client.api.top(self.id, *args, **kwargs)
+
+    def unpause(self, *args, **kwargs):
+        return self.client.api.unpause(self.id, *args, **kwargs)
+
+    def update(self, *args, **kwargs):
+        return self.client.api.update_container(self.id, *args, **kwargs)
+
+    def wait(self, *args, **kwargs):
+        return self.client.api.wait(self.id, *args, **kwargs)
+
+
+class ContainerCollection(Collection):
+    model = Container
+
+    def run(self, image, command=None, stdout=True, stderr=False, remove=False,
+            **kwargs):
+        if isinstance(image, Image):
+            image = image.id
+        detach = kwargs.get("detach", False)
+        if detach and remove:
+            raise RuntimeError("The options 'detach' and 'remove' cannot be "
+                               "used together.")
+        try:
+            container = self.create(image, command, **kwargs)
+        except ImageNotFound:
+            self.client.images.pull(image)
+            container = self.create(image, command, **kwargs)
+
+        container.start()
+
+        if detach:
+            return container
+
+        exit_status = container.wait()
+        if exit_status != 0:
+            stdout = False
+            stderr = True
+        out = container.logs(stdout=stdout, stderr=stderr)
+        if remove:
+            container.remove()
+        if exit_status != 0:
+            raise ContainerError(container, exit_status, command, image, out)
```

Comment:
For run, certainly. This is a convenience for the 80% case where an error is bad. If for some reason you don't want to be handling exceptions, you can still do `client.containers.run(..., detach=True); client.containers.wait()`.

Generally, this is part of a broader direction we're trying to take the SDKs. This is not about running containers as such, this is about making this a toolkit for packaging up part of your app as containers. This allows you to treat containers as functions that could be anywhere, in any language.

For example, suppose you were writing a Python app and you wanted to call a piece of functionality in Node.js called "reticulate-splines":

```
reticulated_splines = client.containers.run('tasks/reticulate-splines', splines)
```

If that fails, it's an error in your application.

Happy to take this discussion offline if you'd like, because it is a much bigger discussion than a GitHub comment. ;)